realm/src/main/java/io/realm/RealmResults.java

```diff
@@ -624,4 +662,125 @@ public void set(E object) {
         @Override
         public void remove() { throw new RealmException("Removing elements not supported."); }
     }
+
+    /**
+     * Swap the table_view pointer used by this RealmResults
+     * mostly called when updating the RealmResults from a worker thread
+     * @param handoverTableViewPointer handover pointer to the new table_view
+     */
+    void swapTableViewPointer (long handoverTableViewPointer) {
+        table = query.importHandoverTableView(handoverTableViewPointer, realm.sharedGroup.getNativePointer());
+        isCompleted = true;
+    }
+
+    /**
+     * Set the Future instance returned by the worker thread, we need this instance
+     * to force {@link #load()} an async query, we use it to determine if the current
+     * RealmResults is a sync or async one
+     * @param pendingQuery pending query
+     */
+    void setPendingQuery (Future<Long> pendingQuery) {
+        this.pendingQuery = pendingQuery;
+        if (isLoaded()) {
+            // the query completed before RealmQuery
+            // had a chance to call setPendingQuery to register the pendingQuery (used btw
+            // to determine isLoaded behaviour)
+            onCompleted();
+        } // else, it will be handled by the Realm#handler
+    }
+
+    /**
+     * Determine if the current RealmResults is obtained synchronously or asynchronously (from
```

Comment:
This sentence seems a bit weird. Calling this doesn't determine that as the choice was made when calling `find/findAsync`. Perhaps something like

"Returns {@code true} if the results are not yet loaded, {@code false} if they are still loading. Synchronous query methods like `findAll()` will always return {@code true}, while asynchronous query methods like `findAllAsync()` will return {@code false} until the results are available."